Port shimming-toolbox documentation

jidicula opened the pull request:
The RTD .md files are copied over from [shimming-toolbox](shimming-toolbox/shimming-toolbox) and converted using pandoc into .rst files. RTD build steps are adapted from [AxonDeepSeg](neuropoly/axondeepseg).

API documentation autogeneration has not yet been included in the RTD config.

Resolves #38 
Related to #2

jidicula committed:
[docs][all docs] Copy docs from shimming-toolbox

**Why this change was necessary**
Most of the existing documentation for shimming-toolbox is not MATLAB-specific, so it's a good time to port over the documentation into this project to keep project development centralized. This ensures that documentation can be updated alongside feature implementations.

**What this change does**
Copies over documentation from shimming-toolbox/shimming-toolbox.

**Additional context/notes/links**

Resolves: #38

jidicula committed:
[docs][installation] Add Miniconda installation instructions

**Why this change was necessary**
Installation instructions previously did not include details about
Miniconda environment setup.

**What this change does**
Adds detailed instructions on how to set up a Miniconda environment
for building a development version of the project.
Modifies the Development section of the README to point to the
installation documentation page.

**Additional context/notes/links**

Resolves: #38
Related to: [AxonDeepSeg installation docs](https://axondeepseg.readthedocs.io/en/latest/documentation.html#installation)

jidicula committed:
[docs][all] Convert Markdown files to .rst

**Why this change was necessary**
.rst files are more compatible with sphinx.

**What this change does**
Converts Markdown files to .rst using pandoc:

`pandoc --from=markdown --to=rst --output=foo.rst fo.md`

**Additional context/notes/links**

Related to: #38

jidicula committed:
[docs][CONTRIBUTING] Move contribution guidelines into doc tree

jidicula committed:
[docs][all docs] Remove old .md doc files

jidicula committed:
[docs][doc build] Add ReadTheDocs build configuration

jidicula committed:
[fix][docs] Fix NOTE directives

**Why this change was necessary**
pandoc conversion did not correctly convert the `!!! note` blocks into
ReST `.. NOTE` directives.

**What this change does**
See title.

jidicula committed:
[docs][index] Remove links to indices and search from TOC

jidicula committed:
[docs][index] Fix page titles

jidicula committed:
[docs][flowchart] Remove extraneous mxgraph container

jidicula committed:
[docs][API] Add API documentation section

**Why this change was necessary**
API documentation improves maintainability of this project.

**What this change does**
Adds a new section to the RTD build for API documentation. The API
documentation is autogenerated using the Sphinx Autodoc extension.

**Additional context/notes/links**

Resolves: #2
Resolves: #38

See also:
https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html
https://brendanhasz.github.io/2019/01/05/sphinx.html#autodoc-extension

jcohenadad reviewed (commented):
i would suggest to call this project `shimming-toolbox` instead of `shimming-toolbox-py`, given that in the short term we will officially switch to the python implementation.

jidicula committed:
[docs][all] Rename shimming-toolbox-py to shimming-toolbox

**Why this change was necessary**
The Python implementation will soon become the maintained distribution
of the software.

**What this change does**
See title.

**Additional context/notes/links**

Related to: #41#pullrequestreview-458521854

jidicula committed:
[docs][RTD config] Fix RTD config to match YAML spec

**What this change does**
Changes tabs to spaces to match YAML spec so RTD build can be
completed successfully.

**Additional context/notes/links**

Related to: https://github.com/shimming-toolbox/shimming-toolbox-py/pull/41#issuecomment-666467327

jcohenadad reviewed (approved):
looks good for a first iteration! thanks 🙏

jidicula committed:
[docs][RTD config] Add pip install to RTD config for API docs

**Why this change was necessary**
RTD cannot build API documentation without pip installing the
package.

**What this change does**
Adds a pip install directive to the RTD configuration.

**Additional context/notes/links**

Adapted from rapptz/discord.py/.readthedocs.yml

jidicula committed:
[docs][RTD config] Add package import to config

**Why this change was necessary**
RTD cannot see the package unless it's imported.

**Additional context/notes/links**

Resolves: https://github.com/shimming-toolbox/shimming-toolbox-py/pull/41#issuecomment-666517789

jidicula committed:
[docs][Sphinx config] Add mock imports to Sphinx config

jidicula committed:
[docs][RTD config] Add version field to RTD config

**Why this change was necessary**
Version must be specified, or RTD defaults to v1, which is
deprecated.

**What this change does**
Specifies RTD v2 for YAML config.

**Additional context/notes/links**

See https://docs.readthedocs.io/en/stable/config-file/v2.html#version

jidicula committed:
[docs][Sphinx config] Add click to mock imports

jidicula committed:
[docs][installation] Add dxm2niix as dependency in docs

jidicula committed:
Merge branch 'master' into issue-38
